engine/session.js

```diff
@@ -726,10 +739,11 @@ class Session {
         length = currentVod.getLiveMediaSequencesCount();
         lastDiscontinuity = currentVod.getLastDiscontinuity();
       }
+      await this._sessionState.set("state", SessionState.VOD_NEXT_INITIATING);
+      await this._sessionState.setCurrentVod(slateVod); 
+
```

Comment:
Better to make sure that setting new current vod is done before setting new mseq values, everywhere